RSDKv5/RSDK/User/Core/UserCore.cpp

```diff
@@ -418,12 +418,14 @@ void RSDK::LoadSettingsINI()
             }
         }
 
-#if !RETRO_USE_ORIGINAL_CODE
-        // using standard allocation here due to mod loader trickery
-        gamePadMappings = new GamePadMappings[gamePadCount];
+        if (gamePadCount) {
+#if RETRO_USE_MOD_LOADER
+            // using standard allocation here due to mod loader trickery
```

Comment:
Would recommend only wrapping the if statement for the C++ allocation. Just AllocateStorage on the original should be OK and accurate.

Reply:
That empty storage would still be the case in original code, and since storage doesn't actively allocate anything in base, it should still be OK and not have a memory leak. If you _really_ want it to not allocate anything, you can add it as nonoriginal code as you said.

Reply:
Went the safe route and added `#if !RETRO_USE_ORIGINAL_CODE` on the if statement just for the sake of not allocating an empty entry.